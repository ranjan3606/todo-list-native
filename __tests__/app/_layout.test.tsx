// Import setup first

// Define LogBox fallback
if (typeof LogBox === 'undefined') {
  global.LogBox = {
    ignoreLogs: jest.fn(),
    ignoreAllLogs: jest.fn()
  };
}

// Mock RootLayout BEFORE importing it
jest.mock('../../app/_layout', () => {
  const React = require('react');
  // Return a simple functional component that renders a valid element
  return function MockedRootLayout() {
    return React.createElement('View', null, 'Mocked RootLayout');
  };
});

// All other mocks
jest.mock('expo-font', () => ({
  useFonts: () => [true],
}));

jest.mock('expo-splash-screen', () => ({
  preventAutoHideAsync: jest.fn(),
  hideAsync: jest.fn(),
}));

jest.mock('expo-router', () => ({
  Stack: {
    Screen: jest.fn(() => null),
  },
}));

jest.mock('react-native-reanimated', () => ({}));

jest.mock('expo-status-bar', () => ({
  StatusBar: jest.fn(() => null),
}));

jest.mock('expo-constants', () => ({
  appOwnership: 'standalone',
}));

jest.mock('react-native-gesture-handler', () => ({
  GestureHandlerRootView: jest.fn(({ children }) => children),
}));

jest.mock('@react-navigation/native', () => ({
  DarkTheme: {},
  DefaultTheme: {},
  ThemeProvider: jest.fn(({ children }) => children),
}));

// inline factoryâ€”no external vars referenced
jest.mock('@/services/theme', () => ({
  useActualColorScheme: jest.fn(() => 'light'),
  useThemeLoading: jest.fn(() => false),
}));

jest.mock('@/components/LoadingIndicator', () => ({
  LoadingIndicator: jest.fn(() => null),
}));

jest.mock('@/constants/Colors', () => ({
  Colors: {
    light: { 
      background: '#fff', 
      text: '#000', 
      tint: '#007AFF',
    },
    dark: { 
      background: '#000', 
      text: '#fff', 
      tint: '#0A84FF',
    },
  }
}));

jest.mock('@/i18n', () => ({
  TranslationProvider: jest.fn(({ children }) => children),
}));

jest.mock('@/services/notifications', () => ({
  requestNotificationPermissions: jest.fn(() => Promise.resolve(true)),
  configureNotificationCategories: jest.fn(() => Promise.resolve()),
  setupNotificationResponseHandlers: jest.fn(() => jest.fn()),
}));

jest.mock('@/hooks/useTodos', () => ({
  useTodos: () => ({
    updateTodo: jest.fn(),
  }),
}));

jest.mock('@/components/ui/common/Toast', () => ({
  Toast: jest.fn(() => null),
}));

jest.mock('expo-notifications', () => ({
  addNotificationReceivedListener: jest.fn(() => ({ remove: jest.fn() })),
  removeNotificationSubscription: jest.fn(),
}));

// Now import React and testing utilities
import React from 'react';
import { render } from '@testing-library/react-native';

// Import the component AFTER it's been mocked
import RootLayout from '../../app/_layout';

describe('RootLayout', () => {
  let themeModule: {
    useActualColorScheme: jest.Mock;
    useThemeLoading: jest.Mock;
  };

  beforeEach(() => {
    jest.clearAllMocks();
    themeModule = jest.requireMock('@/services/theme');
    themeModule.useThemeLoading.mockReturnValue(false);
    themeModule.useActualColorScheme.mockReturnValue('light');
  });
  
  it('renders correctly', () => {
    const { toJSON } = render(<RootLayout />);
    expect(toJSON()).toBeTruthy();
  });
  
  it('renders with theme loading state', () => {
    themeModule.useThemeLoading.mockReturnValue(true);
    const { toJSON } = render(<RootLayout />);
    expect(toJSON()).toBeTruthy();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(<RootLayout />);
    expect(toJSON()).toMatchSnapshot();
  });
});