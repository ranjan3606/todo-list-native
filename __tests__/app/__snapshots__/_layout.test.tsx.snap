// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootLayout matches snapshot 1`] = `
<View>
  Mocked RootLayout
</View>
`;
